src/main/java/org/sagebionetworks/web/client/widget/entity/controller/ProvenanceEditorWidget.java

```diff
@@ -0,0 +1,216 @@
+		this.entityFinder = entityFinder;
+		this.urlDialog = urlDialog;
+		usedProvenanceList = ginInjector.getProvenanceListWidget();
+		executedProvenanceList = ginInjector.getProvenanceListWidget();
+		usedProvenanceList.setEntityFinder(entityFinder);
+		usedProvenanceList.setURLDialog(urlDialog);
+		executedProvenanceList.setEntityFinder(entityFinder);
+		executedProvenanceList.setURLDialog(urlDialog);
+		view.setSynAlertWidget(synAlert);
+		view.setUsedProvenanceList(usedProvenanceList);
+		view.setExecutedProvenanceList(executedProvenanceList);
+		view.setEntityFinder(entityFinder);
+		view.setURLDialog(urlDialog);
+		view.setPresenter(this);
+	}
+	
+	@Override
+	public void configure(EntityBundle entityBundle, EntityUpdatedHandler entityUpdatedHandler) {
+		this.entityUpdatedHandler = entityUpdatedHandler;
+		clear();
+		Entity entity = entityBundle.getEntity();
+		synapseClient.getOrCreateActivityForEntityVersion(entity.getId(), null, new AsyncCallback<Activity>() {
+			
+			@Override
+			public void onFailure(Throwable caught) {
+				synAlert.handleException(caught);
+			}
+			
+			@Override
+			public void onSuccess(Activity result) {
+				activity = result;
+				view.setName(result.getName());
+				view.setDescription(result.getDescription());
+				Set<Used> allProvenance = result.getUsed();
+				List<ProvenanceEntry> usedEntries = new LinkedList<ProvenanceEntry>();
+				List<ProvenanceEntry> executedEntries = new LinkedList<ProvenanceEntry>();
+				for (Used provEntry: allProvenance) {    
+					ProvenanceEntry toAdd;
+					if (provEntry instanceof UsedEntity) {
+						Reference ref = ((UsedEntity)provEntry).getReference();
+						toAdd = ginInjector.getEntityRefEntry();
+						String entityId = ref.getTargetId();
+						Long version = ref.getTargetVersionNumber();
+						((EntityRefProvEntryView)toAdd).configure(entityId, version != null ? version.toString() : null);
+						toAdd.setAnchorTarget(DisplayUtils.getSynapseHistoryToken(entityId, version));
+					} else {
+						UsedURL usedURL = (UsedURL) provEntry;
+						toAdd = ginInjector.getURLEntry();
+						String name = usedURL.getName();
+						String url = usedURL.getUrl();
+						((URLProvEntryView)toAdd).configure(name != null && !name.trim().isEmpty() ? name : usedURL.getUrl(), url);
+						toAdd.setAnchorTarget(url);
+					}
+					
+					if (provEntry.getWasExecuted()) {
+						executedEntries.add(toAdd);
+					} else {
+						usedEntries.add(toAdd);
+					}
+				}
+				usedProvenanceList.configure(usedEntries);
+				executedProvenanceList.configure(executedEntries);
+			}
+		});
+	}
+	
+	@Override
+	public Widget asWidget() {
+		return view.asWidget();
+	}
+	
+	public void show() {
+		clear();
+		view.show();
+	}
+	
+	public void hide() {
+		view.hide();
+	}
+	
+	@Override
+	public void clear() {
+		usedProvenanceList.clear();
+		executedProvenanceList.clear();
+		synAlert.clear();
+		view.clear();
+	}
+
+	@Override
+	public void onSave() {
+		activity.setName(view.getName());
```

Comment:
Nick confirmed that it's ok to have an empty name (which must be true because we create an empty Activity if none exists!), so this is resolved.